pkg/thing/interactors/update_config.go

```diff
@@ -40,13 +41,48 @@ func (i *ThingInteractor) validateConfig(authorization, id string, configList []
 		return ErrSchemaUndefined
 	}
 
+	err = validateSchemaMatch(configList, thing.Schema)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateSchemaMatch(configList []entities.Config, schemaList []entities.Schema) error {
 	for _, c := range configList {
-		for _, s := range thing.Schema {
-			if c.SensorID != s.SensorID {
-				return ErrConfigInvalid
+		for _, s := range schemaList {
+			if c.SensorID == s.SensorID {
+				return validateFlagValue(c, s.ValueType)
 			}
+
+			return ErrConfigInvalid
 		}
 	}
 
 	return nil
```

Comment:
I think the previous comment will imply in some adaptation here

Reply:
Done.